packages/ui/Touchable/PinchZoom.js

```diff
@@ -0,0 +1,172 @@
+					maxY: bounds.bottom - parseInt(computedStyle.paddingBottom)
+				};
+			} else {
+				bounds = {
+					minX: bounds.left,
+					minY: bounds.top,
+					maxX: bounds.right,
+					maxY: bounds.bottom
+				};
+			}
+		}
+
+		this.bounds = bounds;
+	};
+
+	getBoundsCoords = ({x, y}) => {
+		const {maxX, maxY, minX, minY} = this.bounds;
+
+		return {x: clamp(minX, maxX, x) - minX, y: clamp(minY, maxY, y) - minY};
+	};
+
+	updateZoom = (scale) => {
+		const {maxZoom, minZoom} = this.pinchZoomConfig;
+		const newScale = clamp(minZoom, maxZoom, scale);
+
+		if (newScale !== this.scale) {
+			this.scale = newScale;
+			return true;
+		}
+
+		return false;
+	};
+
+	begin = (config, {noResume, onPinchZoom, onPinchZoomEnd, onPinchZoomStart}, coords, node) => {
+		if (!onPinchZoom && !onPinchZoomStart && !onPinchZoomEnd) {
+			return;
+		}
+
+		this.pinchZoomConfig = {
+			...config,
+			node,
+			resume: !noResume
+		};
+
+		this.setContainerBounds(node);
+		const {x: x1, y: y1} = this.getBoundsCoords(coords[0]);
+		const {x: x2, y: y2} = this.getBoundsCoords(coords[1]);
+		const dx = x1 - x2;
+		const dy = y1 - y2;
+
+		this.startDist = Math.sqrt((dx * dx + dy * dy));
+		this.startScale = this.scale;
+
+		this.onPinchZoom = onPinchZoom;
+		this.onPinchZoomStart = onPinchZoomStart;
+		this.onPinchZoomEnd = onPinchZoomEnd;
+
+		if (this.onPinchZoomStart) {
+			this.onPinchZoomStart({
+				type: 'onPinchZoomStart',
+				coords
+			});
+		}
+	};
+
+	// This method will get the `onPinchZoom`, `onPinchZoomEnd`, `onPinchZoomStart` props.
+	updateProps = ({onPinchZoom, onPinchZoomEnd, onPinchZoomStart}) => {
+		// Check `isZooming` gesture is not in progress. Check if gesture exists before updating the references to the `pinchZoomConfig`
+		if (!this.isZooming()) return;
+
+		// This will update the `pinchZoomConfig` with the new value
+		this.onPinchZoom = onPinchZoom;
+		this.onPinchZoomStart = onPinchZoomStart;
+		this.onPinchZoomEnd = onPinchZoomEnd;
+	};
+
+	move = (coords) => {
+		if (!this.isZooming()) return;
+
+		const {scaleTolerance} = this.pinchZoomConfig;
+
+		let {x: x1, y: y1} = this.getBoundsCoords(coords[0]);
+		let {x: x2, y: y2} = this.getBoundsCoords(coords[1]);
+
+		const dx = x1 - x2;
+		const dy = y1 - y2;
+
+		const scale = (Math.sqrt(dx * dx + dy * dy) / this.startDist) * this.startScale;
+
+		if (Math.abs(this.scale - scale) > scaleTolerance && this.onPinchZoom && this.updateZoom(scale)) {
+			this.onPinchZoom({
+				type: 'onPinchZoom',
+				scale: this.scale,
+				coords
+			});
+		}
+	};
+
+	blur = () => {
+		if (!this.isZooming()) return;
+
+		if (!this.pinchZoomConfig.global) {
+			this.end();
+		}
+	};
+
+	end = () => {
+		if (!this.isZooming()) return;
+
+		if (this.onPinchZoomEnd ) {
+			this.onPinchZoomEnd({type: 'onPinchZoomEnd'});
+		}
+
+		this.pinchZoomConfig = null;
+	};
+}
+
+const defaultPinchZoomConfig = {
+	boxSizing: 'border-box',
+	global: false,
+	maxZoom: 4,
+	minZoom: 0.5,
+	scaleTolerance: 0.02
```

Comment:
@0x64 
I think the way to consider the difference in distance is still valid. Because user can only move with the same distance.
How about consider the diffrence of distance?

```
Math.abs(this.startDist - currentDist) > moveTolerance
```

Reply:
I changed `scaleTolerance ` to `moveTolerance`